courses/utils.py

```diff
@@ -95,6 +95,8 @@ def generate_program_certificate(user, program):
         ProgramCertificate (or None if one was not found or created) paired
         with a boolean indicating whether the certificate was newly created.
     """
+    if not program or not user:
+        return None, False
```

Comment:
I think this logic should be removed. `generate_program_certificate` should expect a `User` and a `Program`, and neither value should be None (we even say so in the docstring for this function). Instead of including a new test for `generate_program_certificate`, we should have a new test for the signal to make sure that `generate_program_certificate` isn't called at all when a non-program course is created

Reply:
Just a safe check instead falling in any exception by this method.